dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -462,6 +464,26 @@ default SortedSet<SourceFile> getVersionSourceFiles(Long entryId, Long versionId
         return new TreeSet<>(sourceFiles);
 
     }
+
+    static Optional<Version> determineRepresentativeVersion(Entry entry) {
+        return Optional.ofNullable(entry.getActualDefaultVersion())
+            .or(() -> determineRepresentativeVersion(entry.getWorkflowVersions()));
+    }
+
+    static <V extends Version> Optional<V> determineRepresentativeVersion(Set<V> versions) {
+        Set<String> mainlineVersionNames = Set.of("master", "main", "develop");
+        Stream<V> mainlineVersions = versions.stream().filter(version -> mainlineVersionNames.contains(version.getName()));
+        Stream<V> validVersions = versions.stream().filter(Version::isValid);
+        Stream<V> allVersions = versions.stream();
+        return youngestVersion(mainlineVersions)
+            .or(() -> youngestVersion(validVersions))
+            .or(() -> youngestVersion(allVersions));
+    }
+
+    private static <V extends Version> Optional<V> youngestVersion(Stream<V> versions) {
+        return versions.max(Comparator.comparing(Version::getId));
```

Comment:
I'd probably pick the most recently updated instead of when the version was created in Dockstore, but you could make arguments both ways, and ultimately the user should chose a default version.